rego/rego.go

```diff
@@ -1851,6 +1878,116 @@ func (r *Rego) eval(ctx context.Context, ectx *EvalContext) (ResultSet, error) {
 	return rs, nil
 }
 
+func (r *Rego) evalWasm(ctx context.Context, ectx *EvalContext) (ResultSet, error) {
+
+	var input *interface{}
+	if ectx.parsedInput != nil {
+		i, err := ast.JSON(ectx.parsedInput)
+		if err != nil {
+			return nil, err
+		}
+		input = &i
+	}
+
+	result, err := r.opa.Eval(ctx, opa.EvalOpts{Metrics: r.metrics, Input: input})
+	if err != nil {
+		return nil, err
+	}
+
+	parsed, err := ast.ParseTerm(string(result.Result))
+	if err != nil {
+		return nil, err
+	}
+
+	resultSet, ok := parsed.Value.(ast.Set)
+	if !ok {
+		return nil, fmt.Errorf("illegal result type")
+	}
+
+	if resultSet.Len() == 0 {
+		return nil, nil
+	}
+
+	if resultSet.Len() > 1 {
```

Comment:
❔ I'm really not seeing it -- why is a len > 1 illegal? 🤔

Reply:
I'll remove that. I think I assumed there's only one object in the set which contains the bindings.

Reply:
If you want to put in a test case, any query that generates multiple answers would work. E.g.,

```rego
a = [1,2]; x = a[i]
```